web_m2x_options_manager/models/m2x_create_edit_option.py

```diff
@@ -0,0 +1,168 @@
+        inverse="_inverse_ir_model_name",
+        store=True,
+        string="Model Name",
+    )
+
+    option_create = fields.Selection(
+        [
+            ("none", "Do nothing"),
+            ("remove", "Remove"),
+            ("force_remove", "Force Remove"),
+        ],
+        default="remove",
+        help='Defines behaviour for "Create" option:\n'
+        "* Do nothing: nothing is done\n"
+        "* Remove: option 'Create' is set to False only if not already"
+        " present\n"
+        "* Force Remove: option 'Create' is always set to False",
+        required=True,
+        string="Create Option",
+    )
+
+    option_create_edit = fields.Selection(
+        [
+            ("none", "Do nothing"),
+            ("remove", "Remove"),
+            ("force_remove", "Force Remove"),
+        ],
+        default="remove",
+        help='Defines behaviour for "Create & Edit" option:\n'
+        "* Do nothing: nothing is done\n"
+        "* Remove: option 'Create & Edit' is set to False only if not already"
+        " present\n"
+        "* Force Remove: option 'Create & Edit' is always set to False",
+        required=True,
+        string="Create & Edit Option",
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        self.clear_caches()
+        return super().create(vals_list)
+
+    def write(self, vals):
+        self.clear_caches()
+        return super().write(vals)
+
+    def unlink(self):
+        self.clear_caches()
+        return super().unlink()
+
+    @api.depends("ir_model_id")
+    def _compute_ir_model_name(self):
+        for opt in self:
+            opt.ir_model_name = opt.ir_model_id.model
+
+    def _inverse_ir_model_name(self):
+        ir_model_obj = self.env["ir.model"]
+        for opt in self:
+            # Using limit=1 since there's a SQL constrain to grant uniqueness
+            opt.ir_model_id = ir_model_obj.search(
+                [("model", "=", opt.ir_model_name)], limit=1
+            )
+
+    @api.constrains("ir_model_name", "ir_model_field_name")
+    def _check_model_and_field_names(self):
+        env = self.env
+        for opt in self:
+            mname, fname = opt.ir_model_name, opt.ir_model_field_name
+            if mname not in env:
+                msg = _("'%s' is not a valid model!")
+                raise ValidationError(msg % mname)
+            elif fname not in env[mname]:
+                msg = _("'%s' is not a valid field for model '%s'!")
+                raise ValidationError(msg % (fname, mname))
+
+    @api.constrains("ir_model_field_id")
+    def _check_field_type(self):
+        ttypes = ("many2many", "many2one")
+        if any(o.ir_model_field_id.ttype not in ttypes for o in self):
+            msg = _("Only Many2many and Many2one fields can be chosen!")
+            raise ValidationError(msg)
+
+    @api.constrains("ir_model_name", "ir_model_field_name")
+    def _check_field_model_uniqueness(self):
```

Comment:
why not an sql constrain to make them unique by model and field?

Reply:
@simahawk this way we can show the user an error message with field and model names, while we wouldn't be able to do that with a SQL constrain

Reply:
I agree with @simahawk here. I understand the UX improvement, but it's not worth the query in the loop IMO